### PR TITLE
chore(bindings): Add TaikoData.EthDeposits[] depositsProcessed to blockMetaDataComponents and tests to accompany

### PR DESCRIPTION
**Pull request by RogerLamTd**

### CODE REVIEW

File: bindings/encoding/input_test.go [hunk 1]
```diff
@@ -84,6 +85,50 @@ func TestEncodeProveBlockInvalidInput(t *testing.T) {
...
+		L1Hash:            abcd,
+		Beneficiary:       common.HexToAddress("0x10020FCb72e27650651B05eD2CEcA493bC807Ba4"),
+		Treasury:          common.HexToAddress("0x50081b12838240B1bA02b3177153Bca678a86078"),
+		TxListHash:        abcd,
+		TxListByteStart:   big.NewInt(0),
+		TxListByteEnd:     big.NewInt(1000),
+		GasLimit:          1,
+		MixHash:           abcd,
+		Timestamp:         uint64(1),
+		DepositsProcessed: []bindings.TaikoDataEthDeposit{},
```
Comment (davidtaikocha) [inline bindings/encoding/input_test.go:109]: Seems we only test the `metaData` with an empty `DepositsProcessed` list? Is it necessary to also test it with another one with non-empty `DepositsProcessed` list?

**Review by davidtaikocha — APPROVED** [review]
